src/forms/TimeInput.jsx

```diff
@@ -107,11 +107,14 @@ export class TimeInput extends React.PureComponent {
 	 * 	handler prop, if there is one provided (eg supplied by redux-form or DateTimePicker)
 	 */
 	onTimeInputChange(e) {
-		const { value } = e.target;
+		let { value } = e.target;
+
 		// Time inputs are allowed to return an empty string for invalid input.
 		// We should ignore all invalid input and return immediately
 		// Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#Validation
-		if (value === '') return;
+		if (value === '') {
+			value = '00:00';
```

Comment:
this is a valid value that will basically never be what the user intended the value to be, which doesn't seem like a good user experience. It _sort of_ aligns with what the 'clear' button would intuitively do in firefox, but not really, and it definitely doesn't align with what you would expect to have saved when you accidentally enter `27:00`, so I think it would be best to try a different solution

Reply:
I can't figure out a way around this. We either:
* Let the browser-native clear button do nothing (only affects Firefox)
* Make the clear button reset the time to `12:00am`

I think making the clear button reset to `12:00am` feels way less strange than having it do nothing at all.

FWIW: a user can't type `27:00` into a time input, it will just end up being `02:07 [AM|PM]` because of the way the browser handles time inputs

Reply:
ok - I don't remember the steps to reproduce the original issue, but there was definitely something that could be typed that made the `value` empty. I guess just go with this solution for now and hopefully it doesn't put people into a confusing state.